config/rc.lua

```diff
@@ -588,3 +588,12 @@ end)
 client.connect_signal("focus", function(c) c.border_color = beautiful.border_focus end)
 client.connect_signal("unfocus", function(c) c.border_color = beautiful.border_normal end)
 -- }}}
+
+mousegrabber.run(
+  function(mouse, button)
+    print(mouse["x"], button)
+    if (mouse["x"] < 100) then
+      return false
+    end
+    return true
+  end, "cross")
```

Comment:
You might want to avoid having this by default? Maybe it is time to add an `awesome-x11` like `tests/` directory with a headless wayland server? Assuming the GLib event loop can be started without deadlocking eveything, GDbus should work? This is "all" it takes to inject the tests.

Reply:
Yeah that was for testing that I forgot about, I'll remove that.

Reply:
What I was testing with this was somewhat difficult to test.

There's not really a way to ensure that when this executes the following happens:
1) Any current mouse grabs (e.g. resizing a window) stops
2) Input is no longer sent to any other client but `way-cooler-client`
3) Keyboard input still works however and new clients take focus
4) The cursor does not change (this is not how Wayland works "by default" - normally clients expose their own cursor and so we have to actively ignore it while in this mode)

The tests will be useful in other cases. E.g setting up fake outputs or inputs and ensuring `way-cooler-client` gets it. But it will take a lot of machinery that is tangential to getting this actually working right now. It's not useful to build regression tests before it even works.

I'm not against tests at all, it's just a lot of work and right now I already have a lot on my plate considering I'm the only contributor. Patches welcome.